src/System.Buffers/src/System/Buffers/NativeBufferBucket.cs

```diff
@@ -33,13 +34,26 @@ internal NativeBufferBucket(int elementsInBuffer, int numberOfBuffers)
             }
         }
 
-        ~NativeBufferBucket()
```

Comment:
Finalizer should call Dispose(false). Dispose(bool) should have all cleanup logic. The "false" branch, should do minimum work to release the resources and should not call other managed object. It can call some statics. Ideally it should just make simple pinvoke calls (or equivalent) to release the resources.